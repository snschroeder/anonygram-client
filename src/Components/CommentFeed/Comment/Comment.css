.Comment {
  margin: .4em 1.0em;
  padding: .5em;
  background: #87818129;
  border-radius: 8px;
}

.Comment > * {
  text-align: left;
}

.Comment__span.username {
  padding: .25em 5px 0 0;
  color: black;
}

.Comment__p {
  word-wrap: break-word;  
}

.Comment__div.timestamp {
  text-align: left;
  color: darkgray;
  font-size:  .9em
}