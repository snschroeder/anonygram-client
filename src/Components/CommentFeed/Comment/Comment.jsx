import React from 'react';
import './Comment.css';

export default function Comment (props) {
  const { username, text, timestamp } = props;

  return (
    <div className='Comment'>
      <div className='Comment__div mainContainer'>
        <p className='Comment__p'><span className='Comment__span username bold'>{username}</span>{text}</p>
      </div>
      <div className='Comment__div timestamp italic'>{timestamp}</div>
    </div>
  )
}

