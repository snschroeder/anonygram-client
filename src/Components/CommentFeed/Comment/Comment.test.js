import React from 'react';
import ReactDOM from 'react-dom';
import Comment from './Comment';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('Comment component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Comment />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders elements by default', () => {
    const wrapper = shallow(<Comment />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders elements given props', () => {
    const props = {
      timestamp: 'in 5 hours',
      username: 'test username',
      key: '2d644486-26dc-4ad3-b460-3c6895aa260f',
      text: 'some comment text',
    };
    const wrapper = shallow(<Comment {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
