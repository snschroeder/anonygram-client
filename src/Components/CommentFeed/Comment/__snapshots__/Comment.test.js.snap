// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment component renders elements by default 1`] = `
<div
  className="Comment"
>
  <div
    className="Comment__div mainContainer"
  >
    <p
      className="Comment__p"
    >
      <span
        className="Comment__span username bold"
      />
    </p>
  </div>
  <div
    className="Comment__div timestamp italic"
  />
</div>
`;

exports[`Comment component renders elements given props 1`] = `
<div
  className="Comment"
>
  <div
    className="Comment__div mainContainer"
  >
    <p
      className="Comment__p"
    >
      <span
        className="Comment__span username bold"
      >
        test username
      </span>
      some comment text
    </p>
  </div>
  <div
    className="Comment__div timestamp italic"
  >
    in 5 hours
  </div>
</div>
`;
